cpp/ql/lib/semmle/code/cpp/models/implementations/Iterator.qll

```diff
@@ -579,31 +614,51 @@ private class IteratorAssignmentMemberOperatorModel extends IteratorAssignmentMe
   override predicate parameterEscapesOnlyViaReturn(int index) { index = -1 }
 }
 
+private string beginName() {
+  result = ["begin", "cbegin", "rbegin", "crbegin", "before_begin", "cbefore_begin"]
+}
+
 /**
  * A `begin` member function, or a related function, that returns an iterator.
  */
-class BeginFunction extends MemberFunction {
+class BeginFunction extends Function {
   BeginFunction() {
-    this.hasName(["begin", "cbegin", "rbegin", "crbegin", "before_begin", "cbefore_begin"]) and
-    this.getType().getUnspecifiedType() instanceof Iterator
+    this.getUnspecifiedType() instanceof Iterator and
+    (
+      this.hasName(beginName()) and
+      this instanceof MemberFunction
+      or
+      this.hasGlobalOrStdOrBslName(beginName()) and
+      not this instanceof MemberFunction and
+      this.getNumberOfParameters() = 1
```

Comment:
A function named `begin` with more than 1 argument? Nope. I'm happy to delete that conjunct if you prefer.

Reply:
Apologies, I should have been more explicit. What I meant was the begin function occurring directly in the `std` or `bsl` namespace, instead of being a class member.

Reply:
Ah! That's quite common: https://en.cppreference.com/w/cpp/iterator/begin

Reply:
Thanks for clarifying.